### ISSUE TITLE
MinMaxTypeMatcher can only match innerst subnode

### ISSUE THREAD

**Issue by DerKnecht** [DiUS/pact-jvm#722]
Hello, I'm relativly new to Pact.
I want to verify in my pact, that the provider serves an answer which is containing at least 1 entry with irrelevant subnodes in a node named "feed".

When I understand that right, the MinMaxTypeMatcher works as following with a given XML-Structure:
```
<feed>
 <entry>
   <title>Blah</title>
   <author>MyAwesomeAuthor</author>
 </entry
 <entry>
   <title>Blob</title>
   <author>SomeOtherAuthor</author>
 </entry>
</feed>
```
When I apply a matcher: `matchers.addRule("$.feed", new MinMaxTypeMatcher(1, 10));`
This will apply the matcher on `$.feed.0.entry.0.title, $.feed.0.entry.1.author, $.feed.1.entry.0.title, $.feed.1.entry.1.author`, right? But this is not what I need.

What I want is to verify my feed has a number of specified entries. How can I do that? This may be a little functional for a Contract, but I want to verify that my consumer knows that the provider serves a list of entries with a previously specified maximum of entries which the consumer gives as parameter. I'm not sure that I use this framework correct this way. If that is the case, how can I create a useful Contract? Please help me a little.

**Comment by uglyog** [comment]
To clarify, you are speaking about a contract where the consumer has specified a maximum number (say six), and you want to verify that the response has at most six entries?

**Comment by DerKnecht** [comment]
Yes. The consumer calls URL: https://myprovider.com/myfeed/?limit=6 and then the consumer expects a response with the structure above (a feed with a maximum number of subnodes - six in this case).
